packages/lodestar/src/api/impl/validator/index.ts

```diff
@@ -227,25 +226,19 @@ export function getValidatorApi({
       // the first `MAXIMUM_GOSSIP_CLOCK_DISPARITY` duration of the epoch `tolerantCurrentEpoch`
       // will equal `currentEpoch + 1`
 
-      const duties: routes.validator.AttesterDuty[] = [];
-      for (const validatorIndex of validatorIndices) {
-        const validator = state.validators[validatorIndex];
-        if (!validator) {
-          throw new ApiError(400, `Validator index ${validatorIndex} not in state`);
-        }
-        const duty = assembleAttesterDuty(
-          config,
-          {pubkey: validator.pubkey, index: validatorIndex},
-          state.epochCtx,
-          epoch
-        );
-        if (duty) duties.push(duty);
+      const validators = state.validators.persistent;
+      const duties = state.epochCtx.getCommitteeAssignments(epoch, new Set(validatorIndices));
+      const data: routes.validator.AttesterDuty[] = [];
+      for (const duty of duties) {
+        const validator = validators.get(duty.validatorIndex);
```

Comment:
Why not use the index2pubkey cache?

Reply:
Turns out using the persistent validators array is much faster. Added a benchmark to prove it.